projects/storefrontlib/src/cms-components/organization/budgets/budget-form/budget-form.component.ts

```diff
@@ -56,7 +61,72 @@ export class BudgetFormComponent extends AbstractFormComponent
     this.orgUnitService.loadOrgUnitNodes();
     this.businessUnits$ = this.orgUnitService.getActiveUnitList();
     if (this.budgetData && Object.keys(this.budgetData).length !== 0) {
-      this.form.patchValue(this.budgetData);
+      const localOffset = this.getLocalTimezoneOffset(true);
+      const startDate = this.formatDateStringWithTimezone(
+        this.budgetData.startDate,
+        localOffset
+      );
+      const endDate = this.formatDateStringWithTimezone(
+        this.budgetData.endDate,
+        localOffset
+      );
+
+      this.form.patchValue({
+        ...this.budgetData,
+        startDate: startDate,
+        endDate: endDate,
+      });
     }
   }
+
+  getLocalTimezoneOffset(invert?: boolean): string {
```

Comment:
There is a noticebale cutom converting logic in this file. I'm curious what a problem are we trying to solve?
Did you think about moving this logic to some service?
Or if it's related to OCC, maybe it's better to place it in normalizers/serializers (converters) layer?